component/src/governance/check.rs

```diff
@@ -112,6 +128,16 @@ pub mod stateful {
             );
         }
 
+        // Check that the proposal ID is the correct next proposal ID
+        let next_proposal_id = state.next_proposal_id().await?;
+        if proposal.id != next_proposal_id {
+            anyhow::bail!(
+                "submitted proposal ID {} does not match expected proposal ID {}",
+                proposal.id,
+                next_proposal_id,
+            );
+        }
```

Comment:
Previously the chain decided the proposal ID for the next proposal, because it was in charge: now, because we are self-minting the governance NFTs, we have to specify up-front the proposal we want to propose, and get rejected if it's not the next one in the succession of IDs.